src/planner/expression_binder/returning_binder.cpp

```diff
@@ -9,6 +9,11 @@ ReturningBinder::ReturningBinder(Binder &binder, ClientContext &context) : Expre
 
 BindResult ReturningBinder::BindExpression(unique_ptr<ParsedExpression> *expr_ptr, idx_t depth, bool root_expression) {
 	auto &expr = **expr_ptr;
+	if (expr.GetName() == "rowid") {
```

Comment:
Can this be moved into the binding of the `ColumnRefExpression`? Using the name of a column is rather error-prone, as names might contain aliases which can cause a false positive here.

Alternatively we might want to use `AddGenericBinding` instead of `AddBaseTable` in `BindReturning` so the rowid column is never exposed to begin with.

Reply:
Yes, that looks good. Aliases are misused somewhat also for named parameters - e.g. perhaps you could add a test with the following function `select struct_pack(row_id := 42);`?

The explicit rowid exception is likely not necessary in `ReturningBinder::BindColumnRef` because of the switch to `AddGenericBinding`, but the more explicit/clear error message is nice.

Reply:
Right, I forgot to mention that `AddGenericBinding` throws an error as well, but it's the same error that was thrown before 
```
Binder Error: Referenced column "rowid" not found in FROM clause!
Candidate bindings: "t0.c0"
```
I felt like a user would appreciate a message that rowid isn't supported.